### PR TITLE
Bluetooth: samples: Peripheral and Central UART fix buffer releasing

### PR DESCRIPTION
**Pull request by KAGA164**
This fix issue when sample is attempted to release UART buffer twice.
It contains fix in Zephyr nrf uarte driver

### CODE REVIEW

**Review by grochu — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -134,6 +134,15 @@ Bluetooth samples
 
     * Increased the main stack size from 1024 to 1536 bytes.
 
+  * :ref:`peripheral_uart` sample:
+
+    * Fixed double attempt to free the UART buffer.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:139]: I think we could elaborate this a bit. Is the double attempt something that was in the sample and it is not there anymore? Not sure how to interpret this.

Comment (KAGA164) [reply doc/nrf/releases/release-notes-changelog.rst]: Ok. I wrote the longer paragraph about my changes :)

**Review by kapi-no — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -134,6 +134,19 @@ Bluetooth samples
 
     * Increased the main stack size from 1024 to 1536 bytes.
 
+  * :ref:`peripheral_uart` sample:
+
+    * Fixed double attempt to free the Rx UART buffer.
+      The Rx buffer might have been released twice by main thread.
+      Handling Rx buffer releasing is now fixed in this sample and in the UARTE driver.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:141]:
```suggestion
    * Fixed handling of Rx buffer releasing in this sample and in the UARTE driver.
      Before this fix, the Rx buffer might have been released twice by the main thread.
```
How about this?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -134,6 +134,19 @@ Bluetooth samples
...
+
+  * :ref:`central_uart` sample:
+
+    * Fixed double attempt to free the Rx UART buffer.
+      The Rx buffer might have been released twice by main thread.
+      Handling Rx buffer releasing is now fixed in this sample and in the UARTE driver.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:147]:
```suggestion
    * Fixed handling of Rx buffer releasing in this sample and in the UARTE driver.
      Before this fix, the Rx buffer might have been released twice by the main thread.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/central_uart/src/main.c [hunk 4]
```diff
@@ -210,17 +214,20 @@ static void uart_cb(const struct device *dev, struct uart_event *evt, void *user
...
+		} else {
 			k_free(buf);
-			buf_release = false;
-			current_buf = NULL;
 		}
 
 		break;
 
 	case UART_TX_ABORTED:
+			LOG_DBG("tx_aborted");
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply samples/bluetooth/central_uart/src/main.c]: Good idea @kapi-no. I did it for both samples

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (KAGA164) [reply doc/nrf/releases/release-notes-changelog.rst:142]: I like your proposal. Done

File: doc/nrf/releases/release-notes-changelog.rst (hunk 3 above)
Comment (KAGA164) [reply doc/nrf/releases/release-notes-changelog.rst:148]: Done

**Review by peknis — APPROVED** [review]

**Review by grochu — APPROVED** [review]

**Review by maje-emb — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**